### PR TITLE
side-bar

### PR DESCRIPTION
**Pull request by Samuel270**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Web Experiences/Blueprints/email-capture-corner/.DS_Store
+Web Experiences/Blueprints/.DS_Store
+.DS_Store
+Web Experiences/Blueprints/.DS_Store
```
Comment (bx-conor-oshea) [inline .gitignore]: We need to remove all .DS_Store files in the project before implementing the .gitignore
Also just ".DS_Store" will have the same effect, all other lines are unnecessary.

File: Web Experiences/Blueprints/side-bar/side-bar.css [hunk 2]
```diff
@@ -1,114 +1,147 @@
+/* accomodate every size screen by default */
```
Comment (bx-conor-oshea) [inline Web Experiences/Blueprints/side-bar/side-bar.css:1]: As in alert bar review with spellings and #bx-{{ref}} comment

File: Web Experiences/Blueprints/side-bar/side-bar.js [hunk 3]
```diff
@@ -1,31 +1,45 @@
...
+    let eventToSent = {
         "channel": "WEB",
         "type": "INTERACTION",
         "pos": window._boxever_settings.pointOfSale,
         "browser_id": Boxever.getID(),
         "interactionID":"BX_SIDEBAR",
         "interactionType": interactionType
     };
-    Boxever.eventCreate(eventToSent, function (data) { }, 'json');
+    Boxever.eventCreate(eventToSent, (data)=> { }, 'json');
```
Comment (bx-conor-oshea) [inline Web Experiences/Blueprints/side-bar/side-bar.js]: These interaction events are not standardised across the experiences. I propose the following until we are able to use custom fields in Goal configuration:
```
const sendInteractionToBoxever = (interactionType) =>{
    let eventToSend = {
        "channel": "WEB",
        "type": "[[ Experience ID | String | SIDE_BAR | {required: true}]]_" + interactionType,
        "pos": window._boxever_settings.pointOfSale,
        "browser_id": Boxever.getID()
    };
    Boxever.eventCreate(eventToSend, (data)=> { }, 'json');
    
sendInteractionToBoxever("CLICKED");
```

**Review by bx-conor-oshea — COMMENTED** [review]
Pretty much all comments on these reviews are applicable to all the experiences

File: Web Experiences/Blueprints/side-bar/side-bar.css (hunk 2 above)
Comment (Samuel270) [reply Web Experiences/Blueprints/side-bar/side-bar.css:1]: Done

File: Web Experiences/Blueprints/side-bar/side-bar.js (hunk 3 above)
Comment (Samuel270) [reply Web Experiences/Blueprints/side-bar/side-bar.js]: Done

File: .gitignore (hunk 1 above)
Comment (Samuel270) [reply .gitignore]: I removed the git ignore file, once everything is added to master we immediately check if there is any DS remaining, if none will implement the gitignore file again.

**Review by bx-conor-oshea — APPROVED** [review]
A couple of points for future that I don't think need to hold up this PR:

1. We should send "VIEWED", "CLICKED", and "DISMISSED" for all experiences. "VIEWED" is missing from some.
2. We should probably standardise usage of addEventListener() vs .onclick - I would propose addEventListener across the board but very much open to suggestion

Also the spelling from the Alert Bar PR probably applies to all

**Pull request merged**